File: src/main/java/com/github/theword/queqiao/payload/modle/CommonBaseComponent.java
package com.github.theword.queqiao.payload.modle;

import lombok.Data;

@Data
public class CommonBaseComponent {
    private String text;
    private String color;
    private String font;
    private boolean bold;
    private boolean italic;
    private boolean underlined;
    private boolean strikethrough;
    private boolean obfuscated;
    private String insertion;

    @Override
    public String toString() {
        return this.text;
    }
}

File: src/main/java/com/github/theword/queqiao/tool/payload/modle/component/CommonBaseComponent.kt
package com.github.theword.queqiao.tool.payload.modle.component


open class CommonBaseComponent(
    var text: String,
    var color: String? = null,
    var font: String? = null,
    var bold: Boolean = false,
    var italic: Boolean = false,
    var underlined: Boolean = false,
    var strikethrough: Boolean = false,
    var obfuscated: Boolean = false,
    var insertion: String? = null
) {
    override fun toString(): String {
        return text
    }
}
